src/services/review/ReviewRequestService.ts

```diff
@@ -150,15 +151,65 @@ export default class ReviewRequestService {
     this.apiService
       .getFilesChanged(userWithSiteSessionData.siteName)
       .andThen((filenames) => {
-        // map each filename to its edit metadata
-        const editMetadata = filenames.map((filename) =>
-          this.createEditedItemDtoWithEditMeta(
-            filename,
-            userWithSiteSessionData,
-            stagingLink
-          )
-        )
-        return ResultAsync.combine(editMetadata)
+        if (filenames.length === 0) {
+          return okAsync([])
+        }
+        const filenamesSet = new Set(filenames)
+        return this.apiService
```

Comment:
hmm is this needed? we have a set and a map, but we could use just a map insead here right?